Update MatMulInteger op parsing

gyulaz-htec opened the pull request:
Add missing uint8 input support and zero point tensor support to [MatMulInteger](https://github.com/onnx/onnx/blob/main/docs/Operators.md#MatMulInteger).
For the detailed description see: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/issues/2512

Failing back to `dot` operator when zero point attributes are present, because those can lead out of the limits of uint8 and int8.

TedThemistokleous reviewed (changes requested):
Fix conflicts. Changes for this look good, and I think we're seeing this as part of some other models being read in with the missing uint8 support

gyulaz-htec commented:
> Fix conflicts. Changes for this look good, and I think we're seeing this as part of some other models being read in with the missing uint8 support

@TedThemistokleous I've fixed the conflicts also found the code path with uint8 inputs and no zero points is not covered with test, so I've added a verify test for that.
